src/gluonts/model/tft/_engine.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Optional, List, Callable, Iterator
+import logging
+import tempfile
+import inspect
+import uuid
+import time
+import os
+
+import numpy as np
+import mxnet as mx
+from mxnet.gluon import nn, HybridBlock, Trainer as MXTrainer
+
+from gluonts.mx.trainer import Trainer as BaseTrainer
+from gluonts.mx.trainer.learning_rate_scheduler import MetricAttentiveScheduler
+from gluonts.mx.trainer.model_averaging import (
+    AveragingStrategy,
+    SelectNBestMean,
+    save_epoch_info,
+)
+from gluonts.model.forecast import QuantileForecast
+from gluonts.model.forecast_generator import (
+    QuantileForecastGenerator as BaseForecastGenerator,
+)
+from gluonts.dataset.loader import (
+    DataLoader,
+    TrainDataLoader,
+    ValidationDataLoader,
+    InferenceDataLoader,
+)
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.common import DataEntry
+from gluonts.gluonts_tqdm import tqdm
+from gluonts.support.util import HybridContext
+
+logger = logging.getLogger("gluonts").getChild("trainer")
+MODEL_ARTIFACT_FILE_NAME = "model"
+STATE_ARTIFACT_FILE_NAME = "state"
+LOG_CACHE = set([])
+OutputTransform = Callable[[DataEntry, np.ndarray], np.ndarray]
+
+
+class Trainer(BaseTrainer):
```

Comment:
Why can't use use the `Trainer` defined in `gluonts.mx.trainer`? Or ask differently, what would you need to be able to use it?

Reply:
+1.  In the ConvModel PR, he had also overrode it and said that the base trainer do not accept default argument values in hybrid_forward and overloaded the trainer to give default None to unused feature types.  But I think we can use the `Trainer` directly using boolean or whether to use the features or not in the estimator or directly extend the `Trainer` class.